clmm/theory/generic.py

```diff
@@ -67,3 +69,63 @@ def compute_magnification_bias_from_magnification(magnification, alpha):
         warnings.warn('Magnification is negative for certain radii, \
                     returning nan for magnification bias in this case.')
     return np.array(magnification)**(np.array([alpha]).T - 1)
+def compute_rdelta(mdelta, redshift, cosmo, massdef='mean', delta_mdef=200):
+    # Check massdef
+    if massdef=='mean':
+        rho = cosmo.get_rho_m
+    elif massdef in ('critical', 'virial'):
+        rho = cosmo.get_rho_c
+    else:
+        raise ValueError(f'massdef(={massdef}) must be mean, critical or virial')
+    return ((3.*mdelta)/(4.*np.pi*delta_mdef*rho(redshift)))**(1./3.)
+def compute_profile_mass_in_radius(r3d, redshift, cosmo, mdelta, cdelta,
+                                   massdef='mean', delta_mdef=200,
+                                   halo_profile_model='nfw', alpha=None):
+    rdelta = compute_rdelta(mdelta, redshift, cosmo, massdef, delta_mdef)
+    # Check halo_profile_model
+    if halo_profile_model=='nfw':
+        prof_integ = lambda c: np.log(1. + c) - c/(1. + c)
+    elif halo_profile_model=='einasto':
+        if alpha is None:
+            raise ValueError('alpha must be provided when Einasto profile is selected!')
+        prof_integ = lambda c: gamma(3./alpha)*gammainc(3./alpha, 2./alpha*c**alpha)
```

Comment:
A side note: CCL uses numerical integration to calculate this factor (for the density profiles), while both CT and NC use this analytic form. I don't think it will affect anything, but I would like to bring it up.

Reply:
Yes. I am also wondering why CCL chose to use numerical integration when the analytic expression was available.